src/binary_search_tree/binary_search_tree.js

```diff
@@ -0,0 +1,125 @@
+class Node {
+  constructor(data) {
+    this.data = data
+    this.left = null
+    this.right = null
+  }
+}
+
+class BinarySearchTree {
+  constructor() {
+    this._root = null
+  }
+
+  contains(data) {
+    let foundNode = false
+    let currentNode = this._root
+
+    while(!foundNode && currentNode) {
+      if (data < currentNode.data) {
+        currentNode = currentNode.left
+      } else if (data > currentNode.data) {
+          currentNode = currentNode.right
+      } else {
+          foundNode = true;
+      }
+    }
+    return foundNode
+  }
+
+  insert(data) {
+    let node = new Node(data)
+    let currentNode
+
+    if (this._root === null) {
+      this._root = node
+    } else {
+        currentNode = this._root
+
+        while(true) {
+          if (data < currentNode.data) {
+            if (currentNode.left === null) {
+              currentNode.left = node
+              break
+            } else {
+                currentNode = currentNode.left
+            }
+          } else if (data > currentNode.data) {
+              if (currentNode.right === null) {
+                currentNode.right = node
+                break
+              } else {
+                  currentNode = currentNode.right
+                }
+             } else {
+                break
+          }
+      }
+    }
+  }
```

Comment:
could you possibly compartmentalize some of this logic? Possibly refactor to move some of this logic out of the same function?